engines/misc/src/main/java/org/alfresco/transform/misc/transformers/ImageToPdfTransformer.java

```diff
@@ -0,0 +1,218 @@
+    {
+        final PDPage pdfPage = new PDPage(resolvePdfFormat(pdfFormat));
+        pdfDocument.addPage(pdfPage);
+        final PDImageXObject image = LosslessFactory.createFromImage(pdfDocument, bufferedImage);
+        try (PDPageContentStream pdfPageContent = new PDPageContentStream(pdfDocument, pdfPage))
+        {
+            final PDRectangle pageSize = pdfPage.getMediaBox();
+            final float withRatio = pageSize.getWidth() / image.getWidth();
+            final float heightRatio = pageSize.getHeight() / image.getHeight();
+            final float ratio = Stream.of(withRatio, heightRatio, 1f).min(Comparator.naturalOrder()).get();
+            // find image bottom
+            final float y = pageSize.getHeight() - image.getHeight() * ratio;
+            // drawing starts from bottom left corner
+            pdfPageContent.drawImage(image, 0, y, image.getWidth() * ratio, image.getHeight() * ratio);
+        }
+    }
+
+    private PDRectangle resolvePdfFormat(final String pdfFormat)
+    {
+        switch (pdfFormat.toUpperCase()) {
+        case "A4":
+            return DEFAULT_PDF_FORMAT;
+        case "LETTER":
+            return PDRectangle.LETTER;
+        case "A0":
+            return PDRectangle.A0;
+        case "A1":
+            return PDRectangle.A1;
+        case "A2":
+            return PDRectangle.A2;
+        case "A3":
+            return PDRectangle.A3;
+        case "A5":
+            return PDRectangle.A5;
+        case "A6":
+            return PDRectangle.A6;
+        case "LEGAL":
+            return PDRectangle.LEGAL;
+        default:
+            log.info("PDF format: '{}' not supported. Using default: '{}'", pdfFormat, DEFAULT_PDF_FORMAT_STRING);
+            return DEFAULT_PDF_FORMAT;
+        }
+    }
+
+    private static <T> Optional<T> parseOptionIfPresent(final Map<String, String> transformOptions, final String parameter, final Class<T> targetType)
+    {
+        if (transformOptions.containsKey(parameter))
+        {
+            final String option = transformOptions.get(parameter);
+            if (targetType == Integer.class)
+            {
+                try
+                {
+                    return Optional.of(targetType.cast(Integer.parseInt(option)));
+                }
+                catch (NumberFormatException e)
+                {
+                    throw new IllegalArgumentException(String.format(INVALID_OPTION_ERROR_MESSAGE, parameter, option));
+                }
+            }
+            else
+            {
+                return Optional.of(targetType.cast(option));
+            }
+        }
+
+        return Optional.empty();
+    }
+
+    private static void verifyOptions(final Integer startPage, final Integer endPage)
+    {
+        if (startPage != null && startPage < 0)
+        {
+            throw new IllegalArgumentException(NEGATIVE_START_PAGE_ERROR_MESSAGE);
+        }
+
+        if (endPage != null && endPage < 0)
```

Comment:
What does start or end page of zero mean? Is the first page number 0?

Reply:
Yes, pages starts from 0.